openfisca_us/variables/usda/snap/income/deductions/shelter/snap_excess_shelter_expense_deduction.py

```diff
@@ -34,8 +34,14 @@ def formula(spm_unit, period, parameters):
             has_elderly_disabled, uncapped_ded, capped_ded
         ) + spm_unit("snap_utility_allowance", period)
         # Homeless shelter deduction is flat and has no utility component.
+        state = spm_unit.household("state_code_str", period)
+        homeless_deduction = (
+            p.homeless.deduction * WEEKS_IN_YEAR * p.homeless.available[state]
+        )
         return where(
-            spm_unit.household("is_homeless", period),
-            p.homeless * 12,
+            spm_unit.household("is_homeless", period)
+            & (housing_cost > 0)
```

Comment:
Added this line here because of [the law](https://www.law.cornell.edu/uscode/text/7/2014#e_4_D) and also [this report](https://fns-prod.azureedge.us/sites/default/files/snap/14-State-Options.pdf#page=20).

Reply:
Good catch. For posterity, it's https://www.law.cornell.edu/uscode/text/7/2014#e_6_D_i_II:
>that are not receiving free shelter throughout the month